web_external/js/views/body/DataPanel.js

```diff
@@ -1,61 +1,101 @@
 minerva.views.DataPanel = minerva.views.Panel.extend({
     events: {
+        // TODO namespace.
         'click .add-dataset-to-session': 'addDatasetToSessionEvent',
+        'click .m-upload-local': 'uploadDialog',
         'click .delete-dataset': 'deleteDatasetEvent',
         'click .csv-mapping': 'mapTableDataset',
         'click .dataset-info': 'displayDatasetInfo',
-        'click .configure-choropleth': 'configureChoropleth'
+        'click .m-configure-geo-render': 'configureGeoRender'
     },
 
-    configureChoropleth: function (event) {
+    /**
+     * Display a modal dialog allowing configuration of GeoJs rendering
+     * properties for the selected dataset.
+     */
+    configureGeoRender: function (event) {
         var datasetId = $(event.currentTarget).attr('m-dataset-id');
         var dataset = this.collection.get(datasetId);
-        if (dataset.get('displayed')) {
-            // don't pop up the modal when the dataset is active
+        var geoRenderType = dataset.getGeoRenderType();
+        if (dataset.get('displayed') || geoRenderType === null) {
+            // Don't pop up the modal when the dataset is active,
+            // or it can't be configured.
             return;
         }
-        if (!this.choroplethRenderWidget) {
-            this.choroplethRenderWidget = new minerva.views.ChoroplethRenderWidget({
-                el: $('#g-dialog-container'),
-                dataset: dataset,
-                parentView: this
-            });
-            this.choroplethRenderWidget.render();
-        } else {
-            this.choroplethRenderWidget.setCurrentDataset(dataset);
-        }
-
-    },
-
-    mapTableDataset: function (event) {
-        var datasetId = $(event.currentTarget).attr('m-dataset-id');
-        var dataset = this.collection.get(datasetId);
-        // TODO may want to split out json from csv at some point
-        // TODO standardize on callback or else dual calls of getX and getXData
-
-        var datasetType = dataset.getDatasetType();
-        if (datasetType === 'json' || datasetType === 'mongo') {
-            this.keymapWidget = new minerva.views.KeymapWidget({
-                el: $('#g-dialog-container'),
-                dataset: dataset,
-                parentView: this
-            });
-            this.keymapWidget.render();
-        } else {
-            // assuming csv
-            // list the files of this item
-            var filesCallback = _.bind(function () {
-                this.tableWidget = new minerva.views.TableWidget({
+        if (geoRenderType === 'choropleth') {
```

Comment:
Can there just be a mapping of render types -> widgets to avoid the similar code?